scipy/special/_special.h

```diff
@@ -15,3 +18,39 @@ inline npy_cdouble lambertw_scalar(npy_cdouble zp, long k, double tol) {
     std::complex<double> w = special::lambertw(z, k, tol);
     return npy_cpack(real(w), imag(w));
 }
+
+inline npy_cdouble csinpi(npy_cdouble zp) {
+    std::complex<double> z(npy_creal(zp), npy_cimag(zp));
+    std::complex<double> w = special::sinpi(z);
```

Comment:
Coming back to this. Did we decide against `reinterpret_cast` for some reason? If no, we should do that. If yes, another option is `special::sinpi({npy_creal(zp), npy_cimag(zp)}` without needing the temporary. Just throwing it out there.

Reply:
More interested in keeping it simple, not about efficiency or anything. Whatever you think looks simplest. Same for all the other wrapper functions.

Reply:
I think the discussion ended with 'use naive copying until we "come up with our own `bit_cast`"', since using `reinterpret_cast` breaks the build